learn/cookbooks/docker.md

```diff
@@ -0,0 +1,135 @@
+
+::: warning
+Meilisearch advises against using the `latest` tag. As its name indicates, `latest` will always get the most recent Meilisearch release, which might result in different machines running different images if significant time passes between setting up each of them.
+:::
+
+## Run Meilisearch
+
+After running `docker pull` and installing Meilisearch, you can use `docker run` to launch it:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+```
+
+### Passing instance options
+
+Meilisearch accepts a number of instance options during launch. You can configure instance options in two ways: environment variables and CLI arguments.
+
+#### Passing instance options with environment variables
+
+To pass environment variables to Docker, you must use the `-e` argument. The example below launches Meilisearch with a master key:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -e MEILI_MASTER_KEY='MASTER_KEY'\
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+```
+
+#### Passing instance options with CLI arguments
+
+In the previous examples, Docker implicitly runs Meilisearch. If you want to pass command-line arguments to Meilisearch with Docker, you must explicitly run the `meilisearch` binary:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+  meilisearch --master-key="MASTER_KEY"
+```
+
+## Managing data
+
+When using Docker, your working directory is `/meili_data`. This means the location of your database file is: `/meili_data/data.ms`
+
+By default, data written to a Docker container is not persistent and is wiped every time the container stops running. This data includes your indexes and the documents they store.
+
+To keep your data intact between reboots, specify a dedicated volume by running Docker with the `-v` command-line option:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+```
+
+The example above uses `$(pwd)/meili_data:/meili_data`, which is a directory in the host machine. This is only recommended when developing and prototyping your application.
+
+::: tip
+If you cannot find the `data.ms` file when following instructions in the Meilisearch documentation, look for `/meili_data/data.ms`.
+:::
+
+### Dumps
+
+To export a dump, use the dumps endpoint as described in our dedicated dumps guide. Once the task is complete, you can access the dump file in the `/meili_data` directory inside the volume you configured with `-v`.
+
+To import a dump, use the `--import-dump` command-line option and specify the path to the dump. Make sure the path points to `/meili_data` in a volume reachable by the Docker:
+
+```sh
+docker run -it --rm \
+  -p 7700:7700 \
+  -v $(pwd)/meili_data:/meili_data \
+  getmeili/meilisearch:v0.28
+  meilisearch --import-dump /meili_data/20200813-042312213.dump
```

Comment:
```suggestion
  meilisearch --import-dump dumps/20200813-042312213.dump
```

The dumps are actually thrown in the `dumps` directory.

Reply:
I don't know if this is relevant but you will have to delete `data.ms` before you can import a dump

Reply:
Yes this is important

Reply:
But it's still inside the mounted volume, right? Wouldn't `/dumps` be more accurate in that case?

Reply:
> But it's still inside the mounted volume, right? Wouldn't /dumps be more accurate in that case?

It's either `/meili_data/dumps` or `dumps`, but not `/dumps`.

> I don't know if this is relevant but you will have to delete data.ms before you can import a dump

Maybe instead of doing a `making a dump` part, we should make an `update meilisearch` part that explains how you can import a dump entirely with the deletion of the `data.ms`? 🤔